db/dcrpg/queries.go

```diff
@@ -3278,6 +3278,7 @@ func appendChartBlocks(charts *cache.ChartData, rows *sql.Rows) error {
 			// datasets sized properly.
 			bigwork = big.NewInt(int64(blocks.Chainwork[len(blocks.Chainwork)-1]))
 		}
+		blocks.Height = append(blocks.Height, height)
```

Comment:
Bug fix 7: `Blocks.Height` was never updated, so length validation with it was not possible.